types/react-tree-graph/index.d.ts

```diff
@@ -0,0 +1,59 @@
+import type { FC, HTMLProps, ReactNode } from "react";
+
+type EventListenerWithNodeId<T> = T extends ((this: infer ListenerThis, ev: infer Event) => infer ListenerReturnType)
+    ? (this: ListenerThis, ev: Event, nodeId: string) => ListenerReturnType
+    : T;
+
+type AddNodeIdToElementEvents<T> = {
+    [K in keyof T]: K extends `on${string}` ? EventListenerWithNodeId<T[K]> : T[K];
+};
+
+interface NodeProps {
+    keyProp?: string;
+    labelProp?: string;
+    shape?: "circle" | "image" | "polygon" | "rect";
+    nodeProps?:
+        | AddNodeIdToElementEvents<HTMLProps<SVGCircleElement>>
+        | AddNodeIdToElementEvents<HTMLProps<SVGRectElement>>
+        | AddNodeIdToElementEvents<HTMLProps<SVGImageElement>>
+        | AddNodeIdToElementEvents<HTMLProps<SVGPolygonElement>>;
+    gProps?: AddNodeIdToElementEvents<HTMLProps<SVGGElement>>;
+    pathProps?: AddNodeIdToElementEvents<HTMLProps<SVGPathElement>>;
```

Comment:
`pathProps` has a different method signature to the others, as it adds both the source and target IDs to the event. Something like:

```
type EventListenerWithPathIds<T> = T extends ((this: infer ListenerThis, ev: infer Event) => infer ListenerReturnType)
     ? (this: ListenerThis, ev: Event, sourceNodeId: string, targetNodeId: string) => ListenerReturnType
     : T;
```

Reply:
Yeah noticed it the second I pushed, I'll upload a new commit in a sec